doc/sonic-sub-port-intf-hld.md

```diff
@@ -0,0 +1,483 @@
+| Verify that sub port router interface entry in ASIC_DB has the updated mtu value                       |
+
+## 6.3 Sub port interface removal
+### 6.3.1 Remove an IP address from a sub port interface
+Test shall cover the IP address being an IPv4 address or an IPv6 address.
+
+| Test case description                                                                                  |
+|--------------------------------------------------------------------------------------------------------|
+| Verify that ip address configuration is removed from CONIFG_DB DOT1Q_INTERFACE table                   |
+| Verify that ip address configuration is removed from APPL_DB INTF_TABLE by Intfmgrd                    |
+| Verify that ip address state ok is removed from STATE_DB INTERFACE_TABLE by Intfmgrd                   |
+| Verify that subnet route entry is removed from ASIC_DB                                                 |
+| Verify that ip2me route entry is removed from ASIC_DB                                                  |
+
+### 6.3.2 Remove all IP addresses from a sub port interface
+| Test case description                                                                                  |
+|--------------------------------------------------------------------------------------------------------|
+| Verify that sub port router interface entry is removed from ASIC_DB                                    |
+
+### 6.3.3 Remove a sub port interface
+Test shall cover the parent interface being a physical port or a port channel.
+
+| Test case description                                                                                  |
+|--------------------------------------------------------------------------------------------------------|
+| Verify that sub port interface configuration is removed from CONIFG_DB DOT1Q_INTERFACE table           |
+| Verify that sub port interface configuration is removed from APPL_DB INTF_TABLE by Intfmgrd            |
+| Verify that sub port interface state ok is removed from STATE_DB by Intfmgrd                           |
+
+# 7 Scalability
+Scalability is ASIC-dependent.
+We enforce a minimum scalability requirement on the number of sub port interfaces that shall be supported on a SONiC switch.
+
+| Name                                                              | Scaling value             |
+|-------------------------------------------------------------------|---------------------------|
+| Number of sub port interfaces per physical port or port channel   | 250                       |
+| Number of sub port interfaces per switch                          | 750                       |
+
+# 8 Port channel renaming
+Linux has the limitation of 16 characters on an interface name.
+For sub port interface use cases on port channels, we need to redesign the current naming convention for port channels (PortChannelXXXX, 15 characters) to take shorter names (such as, PoXXXX, 6 characters).
```

Comment:
Physical port such as Ethernet128.1024 will exceed limitation of 16 characters in Linux kernel. It is better to take physical port into account for supporting 128-ports device.